### ISSUE TITLE
IndicesDataStreamsStatsResponse.total_store_size_bytes should be a long

### ISSUE THREAD

**Issue by molsza** [elastic/elasticsearch-specification#2099]
### Elasticsearch Version

8.5.3

### Java Version

11

### OS Version

java

### Problem Description

When I am trying to get info about data streams with more 50GB data using the new elastic client for Java I am getting:
```
co.elastic.clients.json.JsonpMappingException: Error deserializing co.elastic.clients.elasticsearch.indices.DataStreamsStatsResponse: jakarta.json.stream.JsonParsingException: Jackson exception: Numeric value (56984969344) out of range of int (-2147483648 - 2147483647) at [Source: (org.apache.http.nio.entity.ContentInputStream); line: 1, column: 128] (JSON path: total_store_size_bytes) (line no=1, column no=128, offset=-1)
```

### Steps to Reproduce

Example code:
```
ElasticsearchClient elasticsearchClient;
elasticsearchClient.indices().dataStreamsStats(s -> s.name("somename"))
```

**Comment by swallez** [comment]
@molsza I have transferred this issue to the `elasticsearch-specification` repository: it contains the API specification used to generate all client libraries, when where the field type has to be updated.

**Comment by Remcojansen88** [comment]
Also waiting for this fix, so that https://github.com/elastic/elasticsearch-java can use the new specifications

**Comment by pquentin** [comment]
This was fixed in #2440 by @l-trotta and should be available in all 8.13 clients. Thanks, closing!
